# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mdadm/endianess-compile.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- mdadm-2.4.1/mdadm.h	2006-03-29 04:47:18.000000000 +0200
+++ mdadm-2.4.1-fixed/mdadm.h	2006-05-09 15:42:34.000000000 +0200
@@ -72,18 +72,25 @@
 #include	"bitmap.h"
 
 #include <endian.h>
-/* #include "asm/byteorder.h" Redhat don't like this so... */
-#if __BYTE_ORDER == __LITTLE_ENDIAN
-#  include <linux/byteorder/little_endian.h>
-#elif __BYTE_ORDER == __BIG_ENDIAN
-#  include <linux/byteorder/big_endian.h>
-#elif __BYTE_ORDER == __PDP_ENDIAN
-#  include <linux/byteorder/pdp_endian.h>
+
+#ifndef __BIG_ENDIAN__
+#define __cpu_to_le16(x) x
+#define __cpu_to_le32(x) x
+#define __cpu_to_le64(x) x
+#define __le16_to_cpu(x) x
+#define __le32_to_cpu(x) x
+#define __le64_to_cpu(x) x
 #else
-#  error "unknown endianness."
+#include <byteswap.h>
+#define __cpu_to_le16(x) __bswap_16(x)
+#define __cpu_to_le32(x) __bswap_32(x)
+#define __cpu_to_le64(x) __bswap_64(x)
+#define __le16_to_cpu(x) __bswap_16(x)
+#define __le32_to_cpu(x) __bswap_32(x)
+#define __le64_to_cpu(x) __bswap_64(x)
 #endif
 
-
+#
 
 /* general information that might be extracted from a superblock */
 struct mdinfo {
